aws-api-appsync/src/main/java/com/amplifyframework/api/aws/TemporalDeserializers.java

```diff
@@ -30,53 +34,73 @@
  * AppSync temporal scalars</a>.
  */
 public final class TemporalDeserializers {
+    private static final Logger LOG = Amplify.Logging.forNamespace("amplify:aws-datastore");
+
     /**
      * Deserializer of Temporal.Date, an extended ISO-8601 Date string, with an optional timezone offset.
-     *
+     * <p>
      * Based on the <a href=https://docs.aws.amazon.com/appsync/latest/devguide/scalars.html> AWS AppSync AWSDate
      * scalar.</a>
      */
     public static final class DateDeserializer implements JsonDeserializer<Temporal.Date> {
         @Override
         public Temporal.Date deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context)
                 throws JsonParseException {
-            return new Temporal.Date(json.getAsString());
+            try {
+                return new Temporal.Date(json.getAsString());
+            } catch (IllegalArgumentException exception) {
+                LOG.warn("Failed to deserialize " + json.getAsString() + " as a Temporal.Date due to " + exception);
+                return null;
```

Comment:
It would be better to throw exceptions early here instead of returning null. This "erases" information, going from a detailed failure, to a generic `null` state.

Same for other deserializers

Reply:
sure sir
so we need this right?
![image](https://user-images.githubusercontent.com/50016799/93170492-f8aac780-f744-11ea-8fb4-8b2cd26c9353.png)

Reply:
I would suggest rethrowing as a JsonParseException, since that is something that consumers of this class will be expecting.

Reply:
ok understand as its an exception because of wrong parsing

Reply:
that's we want to throw JsonParseException?
![image](https://user-images.githubusercontent.com/50016799/93172248-1c234180-f748-11ea-816e-56e163143744.png)

Reply:
No, you'll want to catch the IllegalArgumentException (not JsonParseException, and then throw a JsonParseException.  And yes, let's do that for all three.

Reply:
ok understand this right
``` 
           try {
                return new Temporal.Date(json.getAsString());
            } catch (IllegalArgumentException exception) {
                throw new JsonParseException("Failed to deserialize " + json.getAsString() + " as a Temporal.Date due to " + exception);
            } ```